src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCrlCache.cs

```diff
@@ -268,13 +293,21 @@ private static string GetCachedCrlPath(string localFileName, bool mkDir=false)
                     {
                         foreach (GeneralNameAsn name in distributionPoint.DistributionPoint.Value.FullName)
                         {
-                            if (name.Uri != null &&
-                                Uri.TryCreate(name.Uri, UriKind.Absolute, out Uri? uri) &&
-                                uri.Scheme == "http")
+                            if (name.Uri != null)
                             {
-                                return name.Uri;
+                                if (Uri.TryCreate(name.Uri, UriKind.Absolute, out Uri? uri) &&
+                                    uri.Scheme == "http")
+                                {
+                                    return name.Uri;
+                                }
+                                else
+                                {
+                                    OpenSslX509ChainEventSource.Log.NonHttpCdpEntry(name.Uri);
```

Comment:
It might be "http" but failed to parse because of some issue later in the uri.  Is that ok?

Reply:
PerfView would just see it as `NonHttpCdpEntry / http://test?q=w`.

Something that takes the message into account would see `Skipping CDP entry 'http://test..?q=w' because the protocol is not HTTP.`, which seems less than ideal, but not too far off the mark.